tests/unit/test_completer.py

```diff
@@ -13,115 +13,121 @@
-                             '--quiet', '--acl', '--storage-class',
-                             '--sse', '--sse-c', '--sse-c-copy-source',
-                             '--sse-c-copy-source-key',
-                             '--sse-c-key',
-                             '--sse-kms-key-id',
-                             '--exclude', '--include',
-                             '--follow-symlinks', '--no-follow-symlinks',
-                             '--cache-control', '--content-type',
-                             '--content-disposition', '--source-region',
-                             '--content-encoding', '--content-language',
-                             '--expires', '--grants', '--only-show-errors',
-                             '--expected-size', '--page-size',
-                             '--metadata', '--metadata-directive',
-                             '--ignore-glacier-warnings']
-                            + GLOBALOPTS)),
-    ('aws s3 cp --quiet -', -1, set(['--no-guess-mime-type', '--dryrun',
-                                     '--recursive', '--content-type',
-                                     '--follow-symlinks', '--no-follow-symlinks',
-                                     '--content-disposition', '--cache-control',
-                                     '--content-encoding', '--content-language',
-                                     '--expires', '--website-redirect', '--acl',
-                                     '--storage-class', '--sse', '--sse-c',
-                                     '--sse-c-copy-source',
-                                     '--sse-c-copy-source-key',
-                                     '--sse-c-key',
-                                     '--sse-kms-key-id',
-                                     '--exclude', '--include',
-                                     '--source-region', '--metadata',
-                                     '--metadata-directive',
-                                     '--grants', '--only-show-errors',
-                                     '--expected-size', '--page-size',
-                                     '--ignore-glacier-warnings']
-                                    + GLOBALOPTS)),
-    ('aws emr ', -1, set(['add-instance-groups', 'add-steps', 'add-tags',
-                          'create-cluster', 'create-default-roles',
-                          'create-hbase-backup', 'describe-cluster',
-                          'describe-step', 'disable-hbase-backups', 'get',
-                          'install-applications', 'list-clusters',
-                          'list-instances', 'list-steps',
-                          'modify-cluster-attributes', 'modify-instance-groups',
-                          'put', 'remove-tags', 'restore-from-hbase-backup',
-                          'schedule-hbase-backup', 'socks', 'ssh',
-                          'terminate-clusters', 'wait']))
-]
+
+def get_completion_sets(driver):
```

Comment:
This method is massive and hard to follow. It's better than the previous implementation in that it doesn't have to be updated as frequently, but it loses some of the declarative nature. If we are going to use this new approach, then can you document this method please? Maybe even breaking this method into smaller methods would help.